src/sqlfluff/core/rules/base.py

```diff
@@ -113,28 +113,32 @@ class LintFix:
 
     """
 
-    def __init__(self, edit_type, anchor: BaseSegment, edit=None):
+    def __init__(
+        self,
+        edit_type: str,
+        anchor: BaseSegment,
+        edit: Optional[Iterable[BaseSegment]] = None,
```

Comment:
This only accepts `Iterable[BaseSegment]` rather than `Union[BaseSegment, Iterable[BaseSegment]]` as before. Think this is cleaner and more well defined.